common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -50,12 +52,11 @@
 
   // These are only updated by readExternal() or read()
   @Nonnull
-  private Object base;
-  private long offset;
+  private MemoryBlock base;
   private int numBytes;
```

Comment:
It's weird to see a `MemoryBlock` and `numBytes` pair, as `MemoryBlock` already has a `length` property. How about we allow projection on `MemoryBlock`? i.e. allow to increase the base offset and decrease the length.

Reply:
Yeah, since I incrementally use `MemoryBlock`, some fields still remains as duplicated.  
I think that `UTF8String.length` can be used projected into `MemoryBlock.length` since `UTF8String` seems immutable.